feature/amazon/src/main/java/com/revenuecat/purchases/amazon/storeProductConversions.kt

```diff
@@ -61,5 +49,76 @@ fun Product.toStoreProduct(marketplace: String): StoreProduct {
     )
 }
 
+internal fun String.extractPrice(
+    currency: Currency
+): Price {
+    debugLog("Received price is $this. Extracting currency and amount.")
+
+    val (priceNumeric, currencySymbol) = this.parsePriceAndCurrencySymbolUsingRegex() ?: 0.0f to currency.symbol
+
+    debugLog("Extracted price is: $priceNumeric. Currency symbol is $currencySymbol")
+
+    var foundCurrencyCode: String? = null
+    if (currencySymbol == currency.symbol) {
+        debugLog("Currency symbol matches Locale's.")
+        foundCurrencyCode = currency.currencyCode
+    } else {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            // We've seen cases the price being as "US$", but the marketplace currency code being "INR"
+            debugLog("Currency symbol is not the same as Locale's. Iterating over all available currencies to find " +
+                "the currency code.")
+            foundCurrencyCode =
+                Currency.getAvailableCurrencies()
+                    .firstOrNull { it.symbol == currencySymbol }?.currencyCode
+        }
+    }
+
+    val currencyCode = foundCurrencyCode ?: run {
+        debugLog("Couldn't determine currencyCode. Setting currencyCode to symbol sent by Amazon")
+        currencySymbol
+    }
+
+    debugLog("Currency code is $currencyCode")
+
+    return Price(
+        currencyCode,
+        priceAmountMicros = priceNumeric.times(MICROS_MULTIPLIER).toLong()
+    )
+}
+
+internal data class Price(
+    val currencyCode: String,
+    val priceAmountMicros: Long
+)
+
+// Explanations about the regexp:
+// \\d+: match the first(s) number(s)
+// [\\.,\\s]: match a "separator": a dot, comma or space
+// \\d+ (the second one): match the number(s) after the separator.
+// The lasts two are englobed in []*, as they can be repeated 0 or n times.
+private val pattern: Pattern = Pattern.compile("(\\d+[[\\.,\\s]\\d+]*)")
+
+internal fun String.parsePriceAndCurrencySymbolUsingRegex(): Pair<Float, String>? {
+    val matcher = pattern.matcher(this)
+    return if (matcher.find()) {
+        val price: String = matcher.group()
+        var convertiblePrice = price.trim()
+        if (price.contains(",")) {
+            convertiblePrice =
+                if (price.contains(".") || price.length - price.replace(",", "").length > 1) {
```

Comment:
`price.length - price.replace(",", "").length > 1` could we pull that out into a function, maybe a string extension? it's a little unclear what that's checking

Reply:
what about the 1,000 case?

Reply:
is this guaranteed to be in US locale? asking because for example in Uruguay, `1,000,000.31` is NaN, it'd be `1.000.000,31`

Reply:
> what about the 1,000 case?
I don't think that's a possible case since that would be formatted as `1,000.00`, but who knows what Amazon would send by looking at that reported broken case... `1,000` is not correctly handled at the moment so I am going to re-think that...

Reply:
So I thought more about this, and I realized I wasn't counting on `1,000` or `1.000`. So I decided to just go back to use a number formatter, which is the right way of doing this and I think it has always worked correctly. So the main change this PR introduces is the way we extract the currency symbol.